File: forza-data-es-kafka-connector-java/src/main/java/project/forzadata/monitoring/ElasticSearchSinkConnector.java
package project.forzadata.monitoring;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import org.apache.kafka.common.config.ConfigDef;
import org.apache.kafka.common.config.ConfigException;
import org.apache.kafka.connect.connector.Task;
import org.apache.kafka.connect.sink.SinkConnector;

public class ElasticSearchSinkConnector extends SinkConnector {

  private Map<String, String> configProperties;

  @Override
  public void start(Map<String, String> props) {
    this.configProperties = props;
    try {
      // 입력 값을 기준으로 config를 생성
      new ElasticSearchSinkConnectorConfig(props);
    } catch (ConfigException e) {
      throw new ConfigException(e.getMessage(), e);
    }
  }

  @Override
  public Class<? extends Task> taskClass() {
    return ElasticSearchSinkTask.class;
  }

  @Override
  public List<Map<String, String>> taskConfigs(int maxTasks) {
    List<Map<String, String>> taskConfigs = new ArrayList<>();
    Map<String, String> taskProps = new HashMap<>(configProperties);

    for (int i = 0; i < maxTasks; i++) {
      taskConfigs.add(taskProps);
    }

    return taskConfigs;
  }

  @Override
  public void stop() {

  }

  @Override
  public ConfigDef config() {
    return ElasticSearchSinkConnectorConfig.CONFIG;
  }

  @Override
  public String version() {
    return "1.0";
  }
}

File: forza-data-es-kafka-connector/src/main/kotlin/project/forzadata/monitoring/ElasticSearchSinkConnector.kt
package project.forzadata.monitoring

import org.apache.kafka.common.config.ConfigDef
import org.apache.kafka.common.config.ConfigException
import org.apache.kafka.connect.connector.Task
import org.apache.kafka.connect.sink.SinkConnector

class ElasticSearchSinkConnector : SinkConnector() {

    private lateinit var configProperties: MutableMap<String, String>

    override fun version(): String {
        return "1.0"
    }

    // 사용자가 입력한 설정값이 들어온다
    override fun start(props: MutableMap<String, String>) {
        this.configProperties = props
        try {
            // 입력 값을 기준으로 config를 생성
            ElasticSearchSinkConnectorConfig(props)
        } catch (e: ConfigException) {
            throw ConfigException(e.message, e)
        }
    }

    override fun taskClass(): Class<out Task> {
        return ElasticSearchSinkTask::class.java
    }

    override fun taskConfigs(maxTasks: Int): MutableList<Map<String, String>> {
        val taskConfigs: MutableList<Map<String, String>> = mutableListOf()
        val taskProps: MutableMap<String, String> = mutableMapOf()
        taskProps.putAll(this.configProperties)

        for (i in 0 until maxTasks) {
            taskConfigs.add(taskProps)
        }

        return taskConfigs
    }

    override fun stop() {
        TODO("Not yet implemented")
    }

    override fun config(): ConfigDef {
        return ElasticSearchSinkConnectorConfig.CONFIG
    }
}